proarc-common/src/main/java/cz/cas/lib/proarc/common/fedora/RemoteStorage.java

```diff
@@ -319,6 +319,18 @@ public void purge(String logMessage) throws DigitalObjectException {
             }
         }
 
+        public void purgeDatastream(String datastream ,String logMessage) throws DigitalObjectException {
+            try {
+                FedoraClient.purgeDatastream(getPid(), datastream).logMessage(qpEncode(logMessage)).execute(client);
+            } catch (FedoraClientException ex) {
+                if (ex.getStatus() == Status.NOT_FOUND.getStatusCode()) {
+                    throw new DigitalObjectNotFoundException(getPid(), ex);
```

Comment:
REST API Fedory vrací:
- 404 pro delete datastream chybějícího objektu 
- 200 pro delete datastream chybějícího datastreamu
- 200 pro delete datastream nechybějícího datastreamu

Tedy je možné rozlišit chybějící objekt, ale není možné rozlišit chybějící datastream od nechybějícího datastreamu. Je tedy nutné vytvářet junit test? Vývoj Fedory 3 a java clienta pro ní je zastavený, takže ke změně chování dojít nemůže.